lib/Dialect/Seq/SeqOps.cpp

```diff
@@ -508,7 +511,12 @@ void FirRegOp::print(::mlir::OpAsmPrinter &p) {
   }
 
   if (auto preset = getPresetAttr()) {
-    p << " preset " << preset.getValue();
+    p << " preset ";
+    const auto &presetVal = preset.getValue();
+    if (presetVal.isNonNegative())
+      p << presetVal;
+    else
+      p << presetVal.zext(presetVal.getBitWidth() + 1);
```

Comment:
This is supposed to change the round-tripping so that negative preset values will no longer be emitted, correct?

Is there a test that could be added for this?

Reply:
Correct. Sorry for the hasty PR, it is really just a hotfix to make round-tripping work again. I did not even notice that there already is a `seq/round-trip.mlir` test.
To be honest, I had to resist the urge of rewriting the preset attribute entirely. Something about bitcasting a raw integer to whatever type the register has just rubs me the wrong way. I would prefer it to be a `TypedAttrInterface` instead of an `APIntAttr`. But I can see that this will create new parsing headaches regarding aggregate values.